middleware/auth.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+)
+
+func isAuthorized(r *http.Request) bool {
+	// TODO: Add configuration to enable this Auth
+	// This configuration should not be as command line option
+	// Because, one glusterd can be run with `--no-auth` and
+	// other glusterd with auth enabled.
+	if strings.HasPrefix(r.RemoteAddr, "127.0.0.1:") {
```

Comment:
Doing string compares solve just this one case easily. The TODO below to allow Auth requests from other IPs would require you to use `net` comparisons anyway.

We should move the whole of Gluster to use CIDR representation and matching for auth. This is the standard. (I need to open an RFE  issue for this). GD2 should as well.

Also, the `net.Parse` functions don't do any resolution AFAIK, so the impact should be negligible.